File: providers/aws/src/test/java/datagen/services/ec2/model/TransitGatewayGenerator.java
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 * #L%
 */
package datagen.services.ec2.model;

import datagen.BaseGenerator;
import datagen.CommonGenerator;
import software.amazon.awssdk.arns.Arn;
import software.amazon.awssdk.services.ec2.model.TransitGateway;
import software.amazon.awssdk.services.ec2.model.TransitGatewayState;

import java.util.List;

public class TransitGatewayGenerator extends BaseGenerator {
    public static String transitGatewayId() {
        return String.format("tgw-%s", faker.random().hex(30));
    }

    public static Arn transitGatewayArn() {
        return Arn.builder()
                  .service("ec2")
                  .region(CommonGenerator.region().id())
                  .accountId(CommonGenerator.accountId())
                  .partition("transit-gateway")
                  .resource(transitGatewayId())
                  .build();
    }

    public static TransitGatewayState transitGatewayState() {
        return fromArray(TransitGatewayState.values());
    }

    public static List<TransitGateway> transitGateways() {
        return transitGateways(faker.random().nextInt(1, 10));
    }

    public static List<TransitGateway> transitGateways(Integer n) {
        return listGenerator(n, TransitGatewayGenerator::transitGateway);
    }

    public static TransitGateway transitGateway() {
        return TransitGateway.builder()
                             .transitGatewayId(transitGatewayId())
                             .transitGatewayArn(transitGatewayArn().toString())
                             .state(transitGatewayState())
                             .ownerId(CommonGenerator.accountId())
                             .description(faker.lorem().sentence())
                             .creationTime(pastDate().toInstant())
                             .options(TransitGatewayOptionsGenerator.transitGatewayOptions())
                             .tags(TagGenerator.tags())
                             .build();
    }
}

File: providers/aws/src/test/kotlin/datagen/services/ec2/model/TransitGatewayGenerator.kt
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package datagen.services.ec2.model

import datagen.BaseGenerator
import datagen.CommonGenerator
import software.amazon.awssdk.arns.Arn
import software.amazon.awssdk.services.ec2.model.TransitGateway
import software.amazon.awssdk.services.ec2.model.TransitGatewayState

object TransitGatewayGenerator : BaseGenerator() {
    fun transitGatewayId(): String {
        return "tgw-${faker.random().hex(30)}"
    }

    fun transitGatewayArn(): Arn {
        return Arn.builder()
                .service("ec2")
                .region(CommonGenerator.region().id())
                .accountId(CommonGenerator.accountId())
                .partition("transit-gateway")
                .resource(transitGatewayId())
                .build()
    }

    fun transitGatewayState(): TransitGatewayState {
        return valueFromArray(TransitGatewayState.values())
    }

    fun transitGateways(n: Int? = null): List<TransitGateway> {
        return listGenerator(n) { transitGateway() }
    }

    fun transitGateway(): TransitGateway {
        return TransitGateway.builder()
                .transitGatewayId(transitGatewayId())
                .transitGatewayArn(transitGatewayArn().toString())
                .state(transitGatewayState())
                .ownerId(CommonGenerator.accountId())
                .description(faker.lorem().sentence())
                .creationTime(pastInstant())
                .options(TransitGatewayOptionsGenerator.transitGatewayOptions())
                .tags(TagGenerator.tags())
                .build()
    }
}
